src/System.Private.CoreLib/shared/System/BitOps.cs

```diff
@@ -219,13 +218,16 @@ public static int RotateLeft(int value, int offset)
         public static uint Log2(uint value)
         {
             // Log(0) is undefined. Return 32 for input 0, without branching.
```

Comment:
For `Log2` we should probably return `0` for `0` (there is some code that depends on it).

Most of the optimized algorithms I can find also return `0` or have an alternative version that returns `-1`. `32` seems right out.

Reply:
Good catch. I originally coded `Log2` as an implementation detail for TZCNT, where I needed the `0->32` behavior. I can change it to `0` now.